cmd/apprepository-controller/main.go

```diff
@@ -30,29 +35,94 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-var (
-	masterURL         string
-	kubeconfig        string
-	repoSyncImage     string
-	repoSyncCommand   string
-	namespace         string
-	dbURL             string
-	dbUser            string
-	dbName            string
-	dbSecretName      string
-	dbSecretKey       string
-	userAgentComment  string
-	crontab           string
-	reposPerNamespace bool
-)
+type arrayFlags []string
+
+// Config contains al the flags passed through the command line
+// besides, it contains the []LocalObjectReference for the PullSecrets
+// so they can be shared across the jobs
+type Config struct {
+	MasterURL                string
+	Kubeconfig               string
+	RepoSyncImage            string
+	RepoSyncImagePullSecrets arrayFlags
+	ImagePullSecretsRefs     []corev1.LocalObjectReference
+	RepoSyncCommand          string
+	Namespace                string
+	DBURL                    string
+	DBUser                   string
+	DBName                   string
+	DBSecretName             string
+	DBSecretKey              string
+	UserAgentComment         string
+	Crontab                  string
+	ReposPerNamespace        bool
+
+	// Args are the positional (non-flag) command-line arguments.
+	Args []string
+}
+
+// parseFlags parses the command-line arguments provided to the program.
+// Typically os.Args[0] is provided as 'programName' and os.args[1:] as 'args'.
+func parseFlags(progname string, args []string) (config *Config, output string, err error) {
```

Comment:
I'm not sure, but it looks like you've created this function to avoid having the global vars even exist? If so, that's admirable, but above and beyond what I meant with my earlier comment - and may not be worth the extra complexity in setup. I think it's clearer to just use the `flag.Parse()` and is what most go-lang apps do, I'd just (generally) avoid accessing those vars anywhere but in the main.go so that it doesn't create problems later on refactoring work (you'll often see people create their sub packages used for this reason, to ensure the boundary is clear and there's no access to the vars in the main package - but no need for that here yet).

Reply:
Ah, I re-read your comment and see that you extracted it so that this function is testable. I don't mind if you want to leave it, but you might find that when switching to `pflag` (which supports the parsing you want) that there's actually nothing to test (you'll just be creating the `Config` struct with the parsed values). See what you think.